data/transactions/logic/eval.go

```diff
@@ -4690,6 +4693,11 @@ func opJSONRef(cx *EvalContext) error {
 		return fmt.Errorf("key %s not found in JSON text", key)
 	}
 	expectedType := JSONRefType(cx.program[cx.pc+1])
+	fs, ok := jsonRefSpecByField(expectedType)
```

Comment:
@jannotti Optional:  Is it worthwhile and/or prudent to make the highlighted check _before_ parsing + evaluating the JSON?

* I realize the related logic is out-of-band to the PR, so I'm good to resolve as is.  It caught my interest to dig a bit deeper.  
* Rationale:  Since parsing the JSON is liable to be an expensive runtime operation, perform as much upfront validation as possible.

Reply:
Yes, I wanted to do that, but stopped myself because I didn't want to increase the size of the PR.  It would more closely resemble the other similar opcodes if we made that change.

Reply:
@jannotti Gotcha - I can prep a PR post-merge.  I set a reminder.  Resolving.